6 Workbench Development Exercise - Traffic Light Widget/trafficLightWidget-wb/src/com/tridiumuniversity/trafficLightWidget/ui/BTrafficLightView.java

```diff
@@ -0,0 +1,39 @@
+package com.tridiumuniversity.trafficLightWidget.ui;
+
+import com.tridiumuniversity.trafficLightWidget.BTrafficLight;
+import com.tridiumuniversity.trafficLightWidget.BTrafficLightState;
+
+import javax.baja.nre.annotations.AgentOn;
+import javax.baja.nre.annotations.NiagaraType;
+import javax.baja.sys.BComponentEvent;
+import javax.baja.sys.BObject;
+import javax.baja.sys.Context;
+import javax.baja.sys.Sys;
+import javax.baja.sys.Type;
+import javax.baja.workbench.view.BWbComponentView;
+
+@NiagaraType(agent = @AgentOn(types = "trafficLightWidget:TrafficLight"))
+public class BTrafficLightView extends BWbComponentView {
+
+    private BTrafficLightWidget traffic = new BTrafficLightWidget();
+    public BTrafficLightView() {
+        setContent(traffic);
+    }
+
+    @Override
+    protected void doLoadValue(BObject obj, Context cx) {
+        BTrafficLight trafficLight = (BTrafficLight) obj;
+        traffic.updateLightState(trafficLight.getState());
+    }
+
+    @Override
+    public void handleComponentEvent(BComponentEvent event) {
```

Comment:
This is explained in the documentation  for `handleComponentEvent`, but in general you should make sure that the source component of `event` is the type that you're expecting because it's possible for this component to receive events from other components when placed on a Px page.